pkg/query/querier.go

```diff
@@ -208,7 +208,19 @@ func (q *querier) Select(_ bool, hints *storage.SelectHints, ms ...*labels.Match
 	// The querier has a context but it gets canceled, as soon as query evaluation is completed, by the engine.
 	// We want to prevent this from happening for the async storea API calls we make while preserving tracing context.
 	ctx := tracing.CopyTraceContext(context.Background(), q.ctx)
-	ctx, cancel := context.WithTimeout(ctx, q.selectTimeout)
+
+	// Check if per-request timeout is specified
```

Comment:
tiny nit: comments should be full sentences with `.`s at the end.

Reply:
Thanks! This suggested change is taken care.